src/interchange/src/avro/encode.rs

```diff
@@ -312,8 +312,13 @@ impl Encode for AvroEncoder {
         "avro"
     }
 
-    fn encode_key_unchecked(&self, row: Row) -> Vec<u8> {
-        self.encode_key_unchecked(self.key_schema_id.unwrap(), row)
+    fn encode_key_unchecked(&self, row: Row) -> (u64, Vec<u8>) {
+        let buf = self.encode_key_unchecked(self.key_schema_id.unwrap(), row);
+        // Compute a stable hash by ignoring the avro header which might contain a
+        // non-deterministic schema id. The Avro header is 5 bytes.
+        // https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
+        let hash = seahash::hash(&buf[5..]);
```

Comment:
Feels a little bad to add a third piece of code that deals with the magic Avro header. Reusing the existing `confluent::extract_avro_header` would only add a tiny perf hit here.

```suggestion
let (schema_id, payload) = confluent::extract_avro_header(buf);
let hash = seahash::hash(payload);
```

Reply:
done